aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3419,6 +3419,10 @@ Tensor flatten(const Tensor& self, int64_t start_dim, int64_t end_dim) {
 }
 
 Tensor flatten(const Tensor& self, int64_t start_dim, int64_t end_dim, Dimname out_dim) {
+  start_dim = maybe_wrap_dim(start_dim, self.dim());
+  end_dim = maybe_wrap_dim(end_dim, self.dim());
+  TORCH_CHECK(start_dim <= end_dim, "flatten() has invalid args: start_dim cannot come after end_dim");
```

Comment:
Hmm, shouldn't it be a ValueError or and IndexError? Also, please mention start and end dim in the error message

Reply:
Thanks for the suggestion. I think it should be an IndexError or ValueError indeed.

I took the check from the basecase (without the out_dim arg)

https://github.com/pytorch/pytorch/blob/33da8d5c1233e0cedeb90a97ffcd360102f667b7/aten/src/ATen/native/TensorShape.cpp#L3394


I can change it to ValueError with start/end dim in the error message but for consistency, need to also change the one from the basecase.

@malfet Should i go ahead for the change at both places or leave things as is? Thanks

Reply:
same TORCH_CHECK changes required here if go ahead

https://github.com/pytorch/pytorch/blob/33da8d5c1233e0cedeb90a97ffcd360102f667b7/aten/src/ATen/native/metal/ops/MetalReshape.mm#L77

Reply:
Hmm, let's leave it until followups then, because changing those might be considered BC breaking

Reply:
Sure, can you re-check if everything looks good (commited the suggested change)? Thanks @malfet

Reply:
Probably a good idea to open an issue for the TORCH_CHECK level changes for tracking it separately? Thanks!